core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -763,7 +763,7 @@ public boolean append(Object kbase, long koff, int klen, Object vbase, long voff
           if (longArray.size() / 2 < MAX_CAPACITY) {
             try {
               growAndRehash();
-            } catch (SparkOutOfMemoryError oom) {
+            } catch (OutOfMemoryError oom) {
```

Comment:
Ya. This was the root cause of reverting. I want to pass the UT once more due to this.

Reply:
Makes sense. In addition to the CI tests, I just ran the manual test in the PR description before and after this fix and confirmed that the bug was present in 2.4 and that this PR fixes it.

Reply:
Thank you so much for confirming, @ankurdave !